### PR TITLE
BP-43: Gradle migration

### PR DESCRIPTION
**Pull request by pkumar-singh**
### Motivation

This PR is to submit proposal for migrating existing maven based build and release pipeline to a gradle based one.

### Changes
Adding a new BP BP-43 for gradle migration. 
Master Issue: https://github.com/apache/bookkeeper/issues/2640

### CODE REVIEW

**Review by merlimat — APPROVED** [review]

**Comment by hsaputra** [comment]
Will merge this by tomorrow if no one has comment or concern about this. Thank you.

File: site/bps/BP-43-gradle-migration.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+   - circe-checksum
+   - bookkeeper-tools-framework
+   
+  If change is made on any of the above mentioned dependent modules, building module `bookkeeper-server` will pick up those changes. 
+  While in case of maven changed modules had to be built explicitly in order for the changes to be picked up.
+  For instance If a change is made in `bookkeeper-common` with gradle 
+  `./gradlew bookkeeper-server:test` will pick up those changes and run tests based on those changes. 
+  While on the other hands for maven `mvn  install bookkeeper-common` had to run before running mvn build `bookkeeper-server`
+
+_Describe the problems you are trying to solve_
```
Comment (dlg99) [inline site/bps/BP-43-gradle-migration.md:30]: not needed?

File: site/bps/BP-43-gradle-migration.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+### Public Interfaces
+N/A
+### Proposed Changes
+- Every module and sub module which are currently being built using maven would be built using gradle.
+- Run all the unit tests, functional tests etc using gradle.
+- Integrating gradle build to CI/CD pipeline 
+- Build ASF release capability using gradle
+- Remove Maven build
+
+### Compatibility, Deprecation, and Migration Plan
```
Comment (dlg99) [inline site/bps/BP-43-gradle-migration.md:41]: documentation update should be a part of the plan (contribution guide/IDE setup, release guide)

**Review by dlg99 — APPROVED** [review]
looks great

**Review by eolivelli — APPROVED** [review]
Very good.
It would be useful to have a way to compare tests results with the Maven build and ensure that we aren't missing test executions

File: site/bps/BP-43-gradle-migration.md (hunk 2 above)
Comment (pkumar-singh) [reply site/bps/BP-43-gradle-migration.md:41]: Sure. I will add that.

File: site/bps/BP-43-gradle-migration.md (hunk 1 above)
Comment (pkumar-singh) [reply site/bps/BP-43-gradle-migration.md:30]: Well. I can drop a line but Motivation section pretty much covers that. Does not it?

**Comment by pkumar-singh** [comment]
> Very good.
> It would be useful to have a way to compare tests results with the Maven build and ensure that we aren't missing test executions

Included in the `test` segment.

File: site/bps/BP-43-gradle-migration.md (hunk 2 above)
Comment (pkumar-singh) [reply site/bps/BP-43-gradle-migration.md:39]: Added.

**Pull request merged**